katsdpcal/reduction.py

```diff
@@ -857,6 +913,56 @@ def flush_pipeline(ts, parameters, solution_stores):
         The pipeline parameters
     solution_stores : dict of :class:`~.CalSolutionStore`-like
         Solution stores for the capture block, indexed by solution type
+    num_chunks : int, optional
+        Group the frequency channels into this many sections to obtain
+        pointing fits
+
+    Returns
+    -------
+    Beam solutions saved to telstate
     """
-    logger.info('~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~')
-    logger.info('Flushed pipeline')
+    # TODO: num_chunks should enter through paramters
+    num_chunks = 16
+    # Extract bandpass gains from solution stores
+    b_solutions = solution_stores['B_POINTING'].get_range(start_time=0, end_time = time.time()).values
+    # Extract some some commonly used constants from the TS and parameters
+    # Middle time for each dump
+    mid_times = b_solutions.get_range(start_time = 0, end_time= time.time()).times
+    target_str = ts.get_range('cbf_target', et = mid_times[-1])[0][0]
+    target = katpoint.Target(target_str)
+    # Atmospheric conditions
+    pres = ts.get_range('anc_air_pressure', et = mid_times[-1])[0][0]
+    temp = ts.get_range('anc_air_temperature', et = mid_times[-1])[0][0]
+    humi = ts.get_range('anc_air_relative_humidity', et = mid_times[-1])[0][0]
+    ants = parameters['antennas']
+    dump_period = ts['int_time']
+    channel_freqs = parameters['channel_freqs']
+    pols = parameters['pol_ordering']
+    
+    # Calculate offset (x,y) co-ordinates for each pointing
+    offsets = get_offsets(ts, b_solutions, target, mid_times, temp, pres, humi, parameters)
+    # Extract gains per pointing offset, per receptor and per frequency chunk.
+    data_points = pointing.get_offset_gains(b_solutions, offsets, ants, dump_period,
+                     channel_freqs, pols, num_chunks)
+    # Fit primary beams to the gains
+    beams = pointing.beam_fit(data_points, ants, num_chunks, beam_center=(0.5, 0.5))
+    # Save fitted beams as CalSolution with shape (num_chunks, len(pols), len(ants), 5)
+    beam_sol = np.full((num_chunks, len(pols), len(ants), 5), np.nan)
```

Comment:
The default dtype will be float64 and we want float32...
```suggestion
    beam_sol = np.full((num_chunks, len(pols), len(ants), 5), np.nan, dtype=np.float32)
```